tests/integration/targets/json_patch/tasks/tests.yml

```
@@ -0,0 +1,262 @@
+# test code for the json_patch module
+# (c) 2019, Joey Espinosa <jlouis.espinosa@gmail.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+- name: Deploy the test json file
+  copy:
+    src: test.json
+    dest: "{{ output_dir }}/test.json"
+  register: result
+
+- name: Store the checksum of the test json file
+  set_fact:
+    test_json_checksum: "91b69ff162bb0805386ae0d57a30a6b992bc5ecc"
+
+- name: Assert that the test json file was deployed
+  assert:
+    that:
+      - result is changed
+      - result.checksum == test_json_checksum
+      - result.state == "file"
+
+- name: Run an add operation in check mode
+  json_patch:
+    src: "{{ output_dir }}/test.json"
+    dest: "{{ output_dir }}/test2.json"
+    operations:
+      - op: add
+        path: "/foo/quz"
+        value: "toto"
+    check_mode: yes
+  register: check_result1
```

```suggestion
  register: check_result1
```
i'd also add `check_mode: yes` check with an existent file, and then check its checksum wasnt changed.